### PR TITLE
Allow to use keyboard-interactive user authentication and set "keyboard-interactive" connection event

### PR DESCRIPTION
**Pull request by dominiklessel**
To fix the following issue: https://github.com/mscdex/ssh2/issues/604

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -86,6 +86,17 @@ export async function normalizeConfig(givenConfig: ConfigGiven): Promise<Config>
       throw new Error('config.password must be a string')
     }
   }
+
+  if (config.tryKeyboard === true && typeof config.onKeyboardInteractive !== 'function') {
```
Comment (steelbrain) [inline src/helpers.js]: Could we also set the default value of `tryKeyboard` here somewhere?

Comment (dominiklessel) [reply src/helpers.js]: Sure!

File: src/helpers.js [hunk 2]
```diff
@@ -86,6 +86,21 @@ export async function normalizeConfig(givenConfig: ConfigGiven): Promise<Config>
       throw new Error('config.password must be a string')
     }
   }
+
+  if (!Object.prototype.hasOwnProperty.call(config, 'tryKeyboard')) {
```
Comment (steelbrain) [inline src/helpers.js]: Actually we could just make it like this 😛 

```
config.tryKeyboard = !!config.tryKeyboard
```

**Comment by dominiklessel** [comment]
Thanks for merging!

**Pull request merged**